### DISCUSSION TITLE
Showing molecules on a scatter plot

### DISCUSSION THREAD

**Discussion by bertiewooster** [rdkit/rdkit#5853] [Q&A] [answered]
Is there a way to put molecules on a scatter plot? For example, for a set of molecules, to

- plot each molecule's melting point against its molecular weight, and
- have each data point show the molecule's structure, either on the graph or as a pop-up when the user mouses over the data point

I think I saw a tool that could do that; it might be outside the RDKit. (The [RDKit cookbook has an example of showing one molecule on a plot](https://www.rdkit.org/docs/Cookbook.html#on-a-plot), but its position isn't determined by its properties that are being plotted.)

**Comment by apahl** [comment, marked as answer]
Hi,
you could use [mol_frame](https://github.com/apahl/mol_frame) (shameless self-plug) or just have a look at the [scatter](https://github.com/apahl/mol_frame/blob/f6ae64a93ea52c747922858a339fc4461e36fcd7/mol_frame/mol_frame.py#L952) function in there for how to achieve your goal with the `holoviews` plotting library.

KR Axel

**Reply by bertiewooster** [reply]
Thanks! I'd actually noticed your [mol_frame](https://github.com/apahl/mol_frame) package before, but didn't appreciate that it offered scatter plots. Very nice [interactive example of mousing over a data point and a tooltip popping up with the molecule's structure](https://github.com/bertiewooster/mol_frame/blob/master/tutorials/scatter_test.html).

I'm trying it out and having trouble importing the package using the [installation instructions](https://github.com/apahl/mol_frame#installation). I contacted you on LinkedIn because mol_frame doesn't have Discussions, and because this is off-topic for RDKit.

**Reply by bertiewooster** [reply]
Update for those interested: With awesome help from @apahl, I was able to get a [mol_frame](https://github.com/apahl/mol_frame) interactive scatter plot working in a [Juypter Notebook on Google Colab](https://colab.research.google.com/drive/1pWK3h6en6rwnGHIDgbz6M5Yo1eDxy9Rr?usp=sharing). If you mouse over a marker on the graph, the molecular structure will be displayed!

![mol_frame-scatter-plot-mouse-over-molecule](https://user-images.githubusercontent.com/1335522/208002553-afb0c1fa-ea9e-4a18-8442-9ee5a7d86834.png)
